tests/common/plugins/conditional_mark/hooks/pre-commit.py

```diff
@@ -0,0 +1,25 @@
+import re
+import sys
+
+def main():
```

Comment:
Probably a simpler algorithm is just to gather the non-comment lines without whitespace in the first char. Sort the list. Then compare the sorted list with the original one?

Reply:
done